### PR TITLE
l2geth: rollup client batch api

### PR DESCRIPTION
**Pull request by tynes**
**Description**
Adds the batch fetching API to the `RollupClient` so that it can be used to fetch batches from the DTL. Follow up PRs will use this code to sync as the verifier and also to ensure that the L2 transactions match what has been batch submitted as the sequencer.

**Additional context**
This is required for https://github.com/ethereum-optimism/optimism/pull/477

### CODE REVIEW

File: l2geth/rollup/client.go [hunk 1]
```diff
@@ -445,6 +493,61 @@ func (c *Client) SyncStatus() (*SyncStatus, error) {
...
+		return nil, nil, fmt.Errorf("Cannot parse transaction batch response")
+	}
+	return parseTransactionBatchResponse(txBatch, c.signer)
+}
+
+// parseTransactionBatchResponse will turn a TransactionBatchResponse into a
+// Batch and its corresponding types.Transactions
+func parseTransactionBatchResponse(txBatch *TransactionBatchResponse, signer *types.OVMSigner) (*Batch, []*types.Transaction, error) {
+	if txBatch == nil {
+		return nil, nil, nil
```
Comment (gakonst) [inline l2geth/rollup/client.go:536]: Should return an error here?

**Review by gakonst — APPROVED** [review]
LGTM - minor comment

File: l2geth/rollup/client.go (hunk 1 above)
Comment (tynes) [reply l2geth/rollup/client.go:536]: Errors are currently not returned in the case where the element does not exist, but it is possible to do so. The consumer would have to do error comparison for the "element not found" case and then act appropriately. I opted to not do specific error checking client side and just return `nil` but now that I think about it, it feels like more idiomatic go to have a non-nil return value when the error is nil.

https://github.com/ethereum-optimism/optimism/blob/c7bc0cef232a1b3957e05671af8fca2a9ef9d892/l2geth/rollup/sync_service.go#L251-L257

Comment (tynes) [reply l2geth/rollup/client.go:543]: I've updated the error handling to use `errors.Is` for error comparison. This is the idiomatic way of doing things, see: https://golang.org/pkg/errors/

Below is from the docs

----

Is unwraps its first argument sequentially looking for an error that matches the second.
It reports whether it finds a match. It should be used in preference to simple equality checks:

```golang
if errors.Is(err, fs.ErrExist)
```

is preferable to
```golang
if err == fs.ErrExist
```

**Pull request merged**